packages/cozy-harvest-lib/src/components/KonnectorIcon.jsx

```diff
@@ -6,10 +6,10 @@ import AppIcon from 'cozy-ui/transpiled/react/AppIcon'
 // TODO move this to cozy-ui
 class KonnectorIcon extends PureComponent {
   fetchIcon() {
-    const { konnector, client } = this.props
+    const { konnector, konnectorSlug, client } = this.props
     return client.getStackClient().getIconURL({
       type: 'konnector',
-      slug: konnector.slug
+      slug: konnectorSlug || konnector.slug
```

Comment:
nit: I'm not a big fan of having a component named `KonnectorXXX` which can takes something that is not a `konnector` object. 

Do we get somewhere the Slug without having the full object?

Reply:
Yes, the refactor here is to be able for the caller not to have a full konnector object. For example if you have io.cozy.accounts and want to show the icon of their konnector, you do not want to create a fake `{ slug: myAccount.account_type}` "konnector" object.